src/D3MHub.sol

```diff
@@ -219,170 +219,123 @@ contract D3MHub {
-            vat.grab(ilk, _pool, _pool, _vow, 0, int256(diff));
+            uint256 fixArt;
+            unchecked {
+                fixArt = ink - art; // Amount of fees + permissionless DAI paid we will now transform to debt
+            }
+            art = ink;
+            vat.suck(_vow, _vow, fixArt * RAY); // This needs to be done to make sure we can deduct sin[vow] and vice in the next call
+            // No need for `fixArt <= MAXINT256` require as:
+            // MAXINT256 >>> MAXUINT256 / RAY which is already restricted above
+            // Also fixArt should be always <= SAFEMAX (MAXINT256 / RAY)
+            vat.grab(ilk, address(_pool), address(_pool), _vow, 0, int256(fixArt)); // Generating the debt
         }
-    }
 
-    // Ilk Getters
-    /**
-        @notice Return pool of an ilk
-        @param ilk   bytes32 of the D3M ilk
-        @return pool address of pool contract
-    */
-    function pool(bytes32 ilk) external view returns (address) {
-        return address(ilks[ilk].pool);
-    }
-
-    /**
-        @notice Return plan of an ilk
-        @param ilk   bytes32 of the D3M ilk
-        @return plan address of plan contract
-    */
-    function plan(bytes32 ilk) external view returns (address) {
-        return address(ilks[ilk].plan);
-    }
+        // Determine if it needs to unwind or wind
+        uint256 toUnwind;
+        uint256 toWind;
 
-    /**
-        @notice Return tau of an ilk
-        @param ilk  bytes32 of the D3M ilk
-        @return tau sec until debt can be written off
-    */
-    function tau(bytes32 ilk) external view returns (uint256) {
-        return ilks[ilk].tau;
-    }
+        // Determine if it needs to fully unwind due to D3M ilk being caged (but not culled) or plan is not active
+        if (ilks[ilk].tic != 0 || !ilks[ilk].plan.active()) {
+            toUnwind = maxWithdraw;
+        } else {
+            uint256 Line = vat.Line();
+            uint256 debt = vat.debt();
+            uint256 targetAssets = ilks[ilk].plan.getTargetAssets(currentAssets);
 
-    /**
-        @notice Return culled status of an ilk
-        @param ilk  bytes32 of the D3M ilk
-        @return culled whether or not the d3m has been culled
-    */
-    function culled(bytes32 ilk) external view returns (uint256) {
-        return ilks[ilk].culled;
-    }
+            // Determine if it needs to unwind due to:
+            unchecked {
+                toUnwind = _max(
+                                _max(
+                                    art > lineWad ? art - lineWad : 0, // ilk debt ceiling exceeded
+                                    debt > Line ? _divup(debt - Line, RAY) : 0 // global debt ceiling exceeded
+                                ),
+                                targetAssets < currentAssets ? currentAssets - targetAssets : 0 // plan targetAssets
+                            );
```

Comment:
```suggestion
                           );
```
indentation fix

Reply:
Actually I always have issues with these type of cases as if I remove a space it breaks the 4 spaces of the tab.